qupath-core/src/main/java/qupath/lib/common/GeneralTools.java

```diff
@@ -775,7 +782,18 @@ public static boolean deleteFile(File fileToDelete, boolean preferTrash) {
 		return fileToDelete.delete();
 	}
 
-	private static boolean moveToTrash(Desktop desktop, File fileToDelete) {
+	/**
+	 * Attempt to move the provided file or directory to trash. This may fail depending on the current platform.
+	 *
+	 * @param desktop the desktop to use when moving the provided file/directory to trash
+	 * @param fileToDelete the file or directory to move to trash
+	 * @return whether the provided file/directory was moved to trash
+	 * @throws NullPointerException if one of the provided parameters is null
+	 * @throws SecurityException if the calling thread doesn't have the permission to read or move the provided file/directory
```

Comment:
Don't think we need to document `SecurityException`. It is dropped from the [javadoc in Java 24](https://docs.oracle.com/en/java/javase/24/docs/api/java.desktop/java/awt/Desktop.html#moveToTrash(java.io.File)), presumably due to [`SecurityManager` being retired](https://inside.java/2024/12/11/quality-heads-up/).